TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -97,44 +96,36 @@ class BenchmarkParser extends Parser {
 
             if ( isValid ) {
 
-                curBenchmarkNamedRouted = BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant];
-
+                curBenchVariant = BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]	
+                curSearchString = curItr.value;
+                
+                // if outerregex is undefined, all runs should be measured. Parse metrics in every run
+                // if outerregex is defined, any runs before outerregex will be ignored. Parse metrics in warm runs only 
+                if (curBenchVariant.outerRegex !== undefined) {
```

Comment:
we need to check `curBenchVariant` value and make sure it is defined.
```
if (curBenchVariant && curBenchVariant.outerRegex) {
```

Reply:
the content of curBenchVariant is
`curBenchVariant = BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]`
                
and its checked before use in line 82
```
else if[curBenchmarkVariant]]
            || !BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]
            || !(Array.isArray(BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]))) {
                isValid = false;
            }
```

Reply:
` curBenchVariant = BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]];`
this value is checked in line 82 
```
else if (!BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]
            || !BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]
            || !(Array.isArray(BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]))) {
                isValid = false;
            }
```